pkg/deployment/kube_service.go

```diff
@@ -34,18 +34,25 @@ func CreateKubeServices(
 	helper *helper.Helper,
 	labels map[string]string,
 ) error {
-	// We create one Service per port and per node, as the Service configuration states that if we
-	// just add all the compute nodes IPs to one service, the Service will round-robin between them.
-	// Our wanted behaviour is to expose all the compute nodes services at the same time.
+	log := helper.GetLogger()
+
+	// We create only one KubeService per port. All the nodes will be IPs on the endpointslices
+	// This will round-robin requests to the nodes, but it is also useful for Prometheus configuration
 	for _, kubeService := range instance.Spec.Services {
 		_, err := service(kubeService, instance, helper, labels)
 		if err != nil {
 			return err
 		}
 
 		addresses := make([]string, len(nodeSet.Spec.Nodes))
+		i := 0
 		for _, item := range nodeSet.Spec.Nodes {
-			addresses = append(addresses, item.Ansible.AnsibleHost)
+			if len(item.Ansible.AnsibleHost) == 0 {
+				log.Info("Creating Services in an IPAM environment is not supported")
```

Comment:
I don't think this log message is accurate, and I don't think the logic is right. If AnsibleHost is not set, we default the ansible var ansible_host to node.HostName in inventory.go.

Also, AnsibleHost not being set doesn't mean that IPAM is in use. And the log message doesn't make sense even if it did. If we're using IPAM, we'd *know* the IP. So why couldn't we create the service?

We also can't say, this just doesn't work when using IPAM. Most customers will use IPAM.